todoapp/src/components/DeletePopUP.vue

```diff
@@ -0,0 +1,168 @@
+<script setup lang="ts">
+import { defineProps, defineEmits } from 'vue'
+import ExclamationIcon from './Icons/ExclamationIcon.vue'
+import CloseIcon from './Icons/CloseIcon.vue'
+
+const props = defineProps<{ index: number }>()
+const emit = defineEmits<{
+  (e: 'togglePopUp'): void
+  (e: 'removeTask', index: number): void
+}>()
+
+function togglePopUp() {
+  emit('togglePopUp')
+}
+
+function removeTask() {
+  emit('removeTask', props.index)
+}
+</script>
+
+<template>
+  <div class="confirmation-dialog-container">
+    <div class="confirmation-dialog-content">
+      <CloseIcon @toggle-pop-up="togglePopUp" />
+      <ExclamationIcon />
+      <div class="popup-title-text">
+        <p class="popup-title">Confirm Deletion</p>
+        <p class="popup-text">Are you sure you want to delete the selected task?</p>
+      </div>
+      <div class="popup-buttons">
+        <button class="popup-btn delete-btn" @click="removeTask">
+          <p class="btn-text">Delete</p>
+        </button>
+        <button class="popup-btn cancel-btn" @click="togglePopUp">
+          <p class="btn-text">Cancel</p>
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+.confirmation-dialog-container {
+  position: absolute;
+  left: 0;
+  top: 0;
+  bottom: 0;
+  right: 0;
```

Comment:
We have a shorthand for setting the top, left, bottom and right to 0 using [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset) you can achieve the same in one line `inset: 0`